app/controllers/publish/courses/outcome_controller.rb

```diff
@@ -19,30 +30,104 @@ def update
         @errors = errors
         return render :edit if @errors.present?
 
+        @current_qualification = @course.qualification
+        @updated_qualification = params[:course][:qualification]
+
         if @course.update(course_params)
-          course_updated_message('Qualification')
+          handle_qualification_update
+        else
+          handle_update_failure
+        end
+      end
 
+      private
+
+      def current_step
+        :outcome
+      end
+
+      def errors
+        params.dig(:course, :qualification) ? {} : { qualification: ['Select a qualification'] }
+      end
+
+      def handle_qualification_update
+        if undergraduate_degree_with_qts?
+          update_undergraduate_course
+        elsif undergraduate_to_other_qualification?
+          redirect_to funding_type_with_previous_course_path
+        else
           redirect_to(
             details_publish_provider_recruitment_cycle_course_path(
               @course.provider_code,
               @course.recruitment_cycle_year,
               @course.course_code
             )
           )
-        else
-          @errors = @course.errors.messages
-          render :edit
+          course_updated_message('Qualification')
         end
       end
 
-      private
+      def handle_update_failure
+        @errors = @course.errors.messages
+        render :edit
+      end
 
-      def current_step
-        :outcome
+      def undergraduate_degree_with_qts?
+        @updated_qualification == 'undergraduate_degree_with_qts'
       end
 
-      def errors
-        params.dig(:course, :qualification) ? {} : { qualification: ['Select a qualification'] }
+      def undergraduate_to_other_qualification?
+        @current_qualification == 'undergraduate_degree_with_qts' && @updated_qualification != 'undergraduate_degree_with_qts'
+      end
+
+      def update_undergraduate_course
+        @course.update(
+          study_mode: 'full_time',
+          funding_type: 'apprenticeship',
+          can_sponsor_skilled_worker_visa: false,
+          can_sponsor_student_visa: false
+        )
```

Comment:
Let's create a service to assign defaults to a TDA course and use in the three places we need.

Reply:
Let's create a system spec that changes the qualification and asserts the all the fields (degrees included) are erased or set to the TDA default.

Reply:
Done here: https://github.com/DFE-Digital/publish-teacher-training/pull/4223/commits/b29e1ae2e0db3987a6ab15b80a7a478acd1a4956